snap/snapshots.go

```diff
@@ -0,0 +1,92 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snap
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/snapcore/snapd/strutil"
+)
+
+var osOpen = os.Open
+
+type SnapshotOptions struct {
+	ExcludePaths []string `yaml:"exclude"`
+}
+
+func ReadSnapshotYaml(si *Info) (*SnapshotOptions, error) {
+	file, err := osOpen(filepath.Join(si.MountDir(), "meta", "snapshots.yaml"))
+	if os.IsNotExist(err) {
+		return &SnapshotOptions{}, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	return readSnapshotYaml(file)
+}
+
+func ReadSnapshotYamlFromSnapFile(snapf Container) (*SnapshotOptions, error) {
```

Comment:
I think ReadInfoFromSnapFile should call this for validation purposes, otherwise is possible to install a snap that then will fail on trying to snapshot.